landscape/lib/apt/package/facade.py

```diff
@@ -269,15 +269,21 @@ def _get_internal_sources_list(self):
         sources_dir = apt_pkg.config.find_dir("Dir::Etc::sourceparts")
         return os.path.join(sources_dir, "_landscape-internal-facade.list")
 
-    def add_channel_apt_deb(self, url, codename, components=None):
+    def add_channel_apt_deb(self, url, codename, components=None,
+                            trusted=None):
         """Add a deb URL which points to a repository.
 
         @param url: The base URL of the repository.
         @param codename: The dist in the repository.
         @param components: The components to be included.
+        @param trusted: Whether validation shoule be skipped (if local).
         """
         sources_file_path = self._get_internal_sources_list()
-        sources_line = "deb %s %s" % (url, codename)
+        source_options = ""
+        if trusted is not None:
+            trusted_val = "yes" if trusted else "no"
+            source_options = "[ trusted={} ] ".format(trusted_val)
```

Comment:
Could collapse these two lines:
if trusted is not None:
        source_options = "[ trusted={} ] ".format("yes" if trusted else "no")

Reply:
Can we also get an unit test on facade to validate trusted=yes|no behavior.

Reply:
That would bust the 80 char and force me to add a line break… Unless you insist I think it's clean as it is.